File: edupath-back/src/main/java/com/edupath/interfaces/IAlumnoRepository.java
package com.edupath.interfaces;

import com.edupath.model.Alumno;
import com.edupath.model.RegistroAcademico;

import java.util.List;

public interface IAlumnoRepository {
    Alumno buscarPorLegajo(String legajo);
    List<RegistroAcademico> obtenerRegistros(String legajo);
}

File: edupath-front/src/app/repositories/i-alumno.repository.ts
import { Observable } from 'rxjs';
import { Alumno } from '../models/alumno.model';
import { RegistroAcademico } from '../models/registro-academico.model';

export interface IAlumnoRepository {
  buscarPorLegajo(legajo: string): Observable<Alumno>;
  obtenerRegistros(legajo: string): Observable<RegistroAcademico[]>;
}

export const I_ALUMNO_REPOSITORY = 'IAlumnoRepository';
